core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala

```diff
@@ -49,11 +50,17 @@ class SubmissionSupervisor(executionServiceCluster: ExecutionServiceCluster,
     actorOf(SubmissionMonitorActor.props(workspaceName, submissionId, datasource, executionServiceCluster, credential, submissionPollInterval, workbenchMetricsBaseName), submissionId.toString)
   }
 
-  override val supervisorStrategy =
-    OneForOneStrategy(maxNrOfRetries = 3) {
-      case e => {
-        system.log.error(e, "error monitoring submission")
-        Restart
-      }
+  // restart the actor on failure (e.g. a DB deadlock or failed transaction)
+  // if this actor has failed more than 3 times, log each new failure
+  override val supervisorStrategy = {
+    val alwaysRestart: SupervisorStrategy.Decider = {
+      case _ => Restart
     }
+
+    def thresholdFunc(cause: Throwable, count: Int): Unit = {
+      system.log.error(cause, s"error monitoring submission: SubmissionMonitorActor has been restarted $count times")
```

Comment:
the exception is logged as part of the log call (but will change this to LazyLogging)